cypress/integration/checkout/checkout.basic.spec.js

```diff
@@ -1,54 +1,122 @@
+        cy.getInstanceTypes({ limit: 1 }).then((instanceTypes) => {
+          itemData.instanceTypeId = instanceTypes[0].id;
+        });
+        cy.getHoldingTypes({ limit: 1 }).then((res) => {
+          itemData.holdingTypeId = res[0].id;
+        });
+        ServicePoints.createViaApi(servicePoint);
+        defaultLocation = Location.getDefaultLocation(servicePoint.id);
+        checkInResultsData.push(defaultLocation.name);
+        Location.createViaApi(defaultLocation);
+        cy.getLoanTypes({ limit: 1 }).then((res) => {
+          itemData.loanTypeId = res[0].id;
+        });
+        cy.getMaterialTypes({ limit: 1 }).then((res) => {
+          itemData.materialTypeId = res.id;
+          itemData.materialTypeName = res.name;
+          checkInResultsData.push(`${itemData.instanceTitle} (${itemData.materialTypeName})`);
+        });
       })
       .then(() => {
-        cy.getUserServicePoints(Cypress.env('users')[0].id);
-
-        cy.createInstance({
+        InventoryInstances.createFolioInstanceViaApi({
           instance: {
-            instanceTypeId: Cypress.env('instanceTypes')[0].id,
-            title: `Pre-checkout instance ${Number(new Date())}`,
+            instanceTypeId: itemData.instanceTypeId,
+            title: itemData.instanceTitle,
           },
-          holdings: [{
-            holdingsTypeId: Cypress.env('holdingsTypes')[0].id,
-            permanentLocationId: Cypress.env('locations')[0].id,
-            sourceId: source.id,
-          }],
+          holdings: [
+            {
+              holdingsTypeId: itemData.holdingTypeId,
+              permanentLocationId: defaultLocation.id,
+            },
+          ],
           items: [
-            [{
-              barcode: ITEM_BARCODE,
-              missingPieces: '3',
-              numberOfMissingPieces: '3',
+            {
+              barcode: itemData.barcode,
               status: { name: 'Available' },
-              permanentLoanType: { id: Cypress.env('loanTypes')[0].id },
-              materialType: { id: Cypress.env('materialTypes')[0].id },
-            }],
+              permanentLoanType: { id: itemData.loanTypeId },
+              materialType: { id: itemData.materialTypeId },
+            },
           ],
         });
+      })
+      .then((specialInstanceIds) => {
+        itemData.testInstanceIds = specialInstanceIds;
       });
-  });
 
-  it('Basic flow', function () {
-    cy.visit('/checkout');
+    cy.createTempUser(userData.permissions)
+      .then((userProperties) => {
+        userData.username = userProperties.username;
+        userData.password = userProperties.password;
+        userData.userId = userProperties.userId;
+        userData.barcode = userProperties.barcode;
+        userData.firstName = userProperties.firstName;
+      })
+      .then(() => {
+        UserEdit.addServicePointViaApi(servicePoint.id, userData.userId, servicePoint.id);
+
+        cy.login(userData.username, userData.password);
```

Comment:
Please add two parameters to this function including path and waiter. the path should be the path to checkout app. You can import it using the import topMenu from '../../support/fragments/topMenu'; and please find the waiter in the related fradment file